File: framework-log/src/main/java/com/example/framework/log/configuretion/LogRecordConfiguration.java
package com.example.framework.log.configuretion;

import com.example.framework.common.entity.BaseUser;
import com.example.framework.log.entity.bo.OperatorBO;
import com.mzt.logapi.service.IOperatorGetService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Optional;

/**
 * @author MaGuangZu
 * @since 2022-08-19
 */
@Configuration
public class LogRecordConfiguration {

	@Bean
	public IOperatorGetService operatorGetService() {
		var user = new BaseUser();
		user.setUserId("1640600092347");
		return () -> Optional.of(user)
			.map(item -> new OperatorBO(item.getUserId()))
			.orElseThrow(() -> new IllegalArgumentException("user is null"));
	}

}

File: framework-log/src/main/kotlin/com/example/framework/log/config/LogRecordConfiguration.kt
package com.example.framework.log.config

import com.example.framework.common.entity.BaseUser
import com.example.framework.log.entity.OperatorDO
import com.mzt.logapi.service.IOperatorGetService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.util.*

/**
 * @author MaGuangZu
 * @since 2022-08-19
 */
@Configuration
open class LogRecordConfiguration {

	@Bean
	open fun operatorGetService(): IOperatorGetService {
        return IOperatorGetService {
            val user = BaseUser()
            user.userId = "1640600092347"
            Optional.of(user)
                .map { item: BaseUser -> OperatorDO(item.userId) }
                .orElseThrow { IllegalArgumentException("user is null") }
        }
    }

}
